pkg/registration/offline_request.go

```diff
@@ -0,0 +1,111 @@
+package registration
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"io"
+	"strings"
+
+	"github.com/SUSE/connect-ng/pkg/connection"
+)
+
+// Encodes the offline requests product which can be activated
+type OfflineRequestProductTriplet struct {
+	Identifier string `json:"identifier"`
+	Version    string `json:"version"`
+	Arch       string `json:"arch"`
+}
+
+// This structure is the opaque version of the base64 encoded offline registration request.
+// It includes all necessary and optional information required to created and activate a system
+// and retrieve an offline registration certificate either via the API [registration.RegisterWithOfflineRequest] or
+// via the [SCC UI]: https://scc.suse.com/register-offline/rancher
+type OfflineRequest struct {
```

Comment:
👍  Looks good. Since we can Marshal to json we can save this to a secret easily. Then that can replace the code I've added to handle this: https://github.com/mallardduck/rancher/blob/17b72246f68c9278a5a43e5119b736639c81bb5a/pkg/scc/systeminfo/exporter.go#L27 & https://github.com/mallardduck/rancher/blob/17b72246f68c9278a5a43e5119b736639c81bb5a/pkg/scc/systeminfo/exporter.go#L103